File: apps/mobile/ios/Uniswap/Onboarding/Import/SeedPhraseInputManager.m
//
//  SeedPhraseInputManager.swift
//  Uniswap
//
//  Created by Gary Ye on 9/7/23.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(SeedPhraseInputManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(targetMnemonicId, NSString?)
RCT_EXPORT_VIEW_PROPERTY(strings, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(onInputValidated, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMnemonicStored, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPasteStart, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPasteEnd, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onSubmitError, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onHeightMeasured, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(testID, NSString?)
RCT_EXTERN_METHOD(handleSubmit: (nonnull NSNumber *)node)
RCT_EXTERN_METHOD(focus: (nonnull NSNumber *)node)
RCT_EXTERN_METHOD(blur: (nonnull NSNumber *)node)

@end

File: apps/mobile/ios/Uniswap/Onboarding/Import/SeedPhraseInputManager.swift
//
//  SeedPhraseInputManager.swift
//  Uniswap
//
//  Created by Gary Ye on 9/15/23.
//

// Using a view manager written in Swift instead of bridging headers
// because couldn't get RCT_EXTERN_METHOD to work with that approach
@objc(SeedPhraseInputManager)
class SeedPhraseInputManager: RCTViewManager {

  override func view() -> UIView! {
    return SeedPhraseInputView()
  }
  
  // Required by RN to initialize on main thread
  override class func requiresMainQueueSetup() -> Bool {
    true
  }
  
  @objc func handleSubmit(_ node: NSNumber) -> Void {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as? SeedPhraseInputView
      component?.handleSubmit()
      // TODO garydebug add error logging for view not found
    }
  }
}
